### PR TITLE
Add SVG support

### PR DESCRIPTION
**Pull request by seagreen**
Add Concur.Replica.SVG and Concur.Replica.SVG.Props, both auto-generated.

### CODE REVIEW

**Comment by seagreen** [comment]
Relevant issue: https://github.com/pkamenarsky/concur-replica/issues/14

Thoughts on the auto-generation strategy, and if we should adopt it for elements as well as attributes?

There aren't nearly as many [elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Element) as there are [attributes](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute), so it would definitely be doable to keep filling them out by hand. That also makes it easier to add some docs in, at minimum like the headers I currently have.

Also a naming question, should the `*Props` modules actually be called `*Attributes`? We could also rethink the naming scheme, because `DOM`, `Props`, `Events`, `SVGProps`, `SVG`, and (a future) `SVGEvents` is a little awkward.

**Comment by pkamenarsky** [comment]
I like the autogeneration idea! So far, I had always copy & pasted element and props definitions from other projects and then made adjustments using e.g. a vim macro, but that's definitely a better way to do it.

> That also makes it easier to add some docs in, at minimum like the headers I currently have.

How about something like:

```
type Header = Text

elementList :: [(Header, [Text])]
elementList =
  [ ( "Animation",
       [ "animate"
       , "animateColor"
       , ...
       ]
     )
   , ...
  ]
```

> Also a naming question, should the *Props modules actually be called *Attributes

My reasoning is that `Props` is shorter but also React and most Purescript UI libraries I've encountered use `Props` instead of `Attributes`.

> We could also rethink the naming scheme, because DOM, Props, Events, SVGProps, SVG, and (a future) SVGEvents

How about this:

```
DOM.hs
DOM/Events.hs
DOM/Props.hs
SVG.hs
SVG/Props.hs
SVG/Events.hs
```

**Comment by seagreen** [comment]
I think all of that is great. Once I get it done I'll take the "WIP" off the PR.

**Comment by seagreen** [comment]
Ready for real review.

A few updates:

+ I switched the SVG modules over to the new scheme, (`Concur.Replica.SVG` and `Concur.Replica.SVG.Props`), but I'll leave the `DOM*` changes for a later PR, just in case merging this one gets delayed and there are changes to those modules on `master` in the meantime.

+ I decided not to organize the SVG module, other than alphabetically. Is that OK? I figured that if people want organized docs MDN will always be better, and for the quick check they might want to do of our module a flat alphabetical structure would be fine.

+ I didn't add an Events module. We can always do that later, and the DOM events for `onClick`, `onMouseDown`, etc work fine.

Thoughts?

File: misc/gen-svg-props.hs [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+          | capitalizeNext       = (False, T.snoc acc (Char.toUpper c))
+          | otherwise            = (False, T.snoc acc c)
+      in
+        snd (T.foldl' f (False, mempty) t)
+
+    toHaskellName :: Text -> Text
+    toHaskellName t =
+      case camelCase t of
+        "class" ->
+          "class'"
```
Comment (seagreen) [inline misc/gen-svg-props.hs]: Slight inconsistency here where DOM uses `className` and this uses `class'`. I could change this to `className`, but that would then make it inconsistent with the ones below.

Comment (pkamenarsky) [reply misc/gen-svg-props.hs]: I propose we keep it in sync with `DOM.Props` here:

* `class` -> `className`
* `type` -> `type_` (see [here](https://github.com/pkamenarsky/concur-replica/blob/master/src/Concur/Replica/Props.hs#L324))
* `in` -> `in_`

Other than that, no complaints!

Comment (seagreen) [reply misc/gen-svg-props.hs]: I like it-- done.

**Comment by pkamenarsky** [comment]
LGTM, thanks for the PR!

**Comment by seagreen** [comment]
You're welcome, it was a pleasure.

**Pull request merged**